library/spdm_secured_message_lib/libspdm_secmes_session.c

```diff
@@ -306,42 +306,54 @@ spdm_generate_session_handshake_key(IN void *spdm_secured_message_context,
                hash_size);
     DEBUG((DEBUG_INFO, "\n"));
 
-    spdm_generate_finished_key(
+    status = spdm_generate_finished_key(
```

Comment:
Issue for a separate PR, but would we ever want to consider scrubbing of session information on failure to calculate secrets, as in here? that way, we don't have half calculated or incorrect secrets in the session context structure?

Reply:
I think we had better to clear the session info.

Reply:
Yeah it is. We can leave it as responsibility of caller to clear, in case there's ever desire to retry versus clear session.